crates/ark/src/variables/variable.rs

```diff
@@ -1537,6 +1565,17 @@ fn parse_index(x: &String) -> harp::Result<isize> {
     })
 }
 
+// We need to be careful when truncating the string, we don't want to return invalid
+// UTF8 sequences. `chars` makes sure we are not splitting a UTF8 character in half.
+// See also https://doc.rust-lang.org/book/ch08-02-strings.html#slicing-strings
+fn truncate_chars(value: String, len: usize) -> (bool, String) {
```

Comment:
IIUC `String::truncate()`would be unsafe here because it takes a byte offset, not a number of chars? I see in the docs:

> Panics if new_len does not lie on a char boundary.

Tricky!